spec/jobs/performance_platform_transactions_queue_job_spec.rb

```diff
@@ -14,15 +24,24 @@
     expect(PerformancePlatformTransactionsQueueJob.new.queue_name).to eq('performance_platform')
   end
 
-  it 'executes perform' do
+  it 'enqueues a job to send all published job from yesterday' do
     stub_const('PP_TRANSACTIONS_BY_CHANNEL_TOKEN', 'not-nil')
 
     pp = double(:performance_platform)
+    two_days_ago = Date.current.beginning_of_day.in_time_zone - 2.days
+    today = Date.current.beginning_of_day.in_time_zone
+
+    build(:vacancy, :past_publish, publish_on: two_days_ago).save(validate: false)
+    build(:vacancy, :past_publish, publish_on: today).save(validate: false)
 
-    published = create_list(:vacancy, 3, :published, publish_on: date)
+    jobs_published_yesterday = [
+      build(:vacancy, :past_publish, publish_on: date_to_upload).save(validate: false),
+      build(:vacancy, :past_publish, publish_on: date_to_upload).save(validate: false)
+    ]
 
     expect(PerformancePlatform::TransactionsByChannel).to receive(:new).with('not-nil').and_return(pp)
-    expect(pp).to receive(:submit_transactions).with(published.count, date.utc.iso8601)
+    expect(pp).to receive(:submit_transactions).with(jobs_published_yesterday.count, date_to_upload.iso8601)
```

Comment:
Yeah, that's a good shout, gives us more confidence. I'll have a look

Reply:
Hmmm... Doesn't `and_return` just specify a return value, rather than expect a return value? I think a better way to do this would be break it out into a method and test that method.

Reply:
TBH, I'm not a fan of this much logic in a job anyway. This really should be in a testable class